app/src/main/java/me/saket/dank/widgets/IndentedLayout.java

```diff
@@ -54,31 +95,61 @@ public IndentedLayout(Context context, @Nullable AttributeSet attrs) {
   protected void onLayout(boolean changed, int l, int t, int r, int b) {
     super.onLayout(changed, l, t, r, b);
 
-    indentationLinePath.reset();
     for (int i = 0; i < indentationDepth; i++) {
       float lineStartX = spacePerDepthPx * (i + 1) + indentationLinePaint.getStrokeWidth();
-      indentationLinePath.moveTo(lineStartX, 0);
-      indentationLinePath.lineTo(lineStartX, getHeight());
+
+      ColoredTree tree = trees.get(i);
+
+      tree.path.reset();
+      tree.path.moveTo(lineStartX, 0);
+      tree.path.lineTo(lineStartX, getHeight());
     }
   }
 
   @Override
   public void draw(Canvas canvas) {
     super.draw(canvas);
-    canvas.drawPath(indentationLinePath, indentationLinePaint);
+
+    for (ColoredTree tree : trees) {
+      indentationLinePaint.setColor(tree.color);
+      canvas.drawPath(tree.path, indentationLinePaint);
+    }
   }
 
   public void setIndentationDepth(@IntRange(from = 0, to = 1) int depth) {
```

Comment:
`@IntRange` shouldn't be there. We can fix that while we are editing this file :)